### DISCUSSION TITLE
Allow extra unspecified query parameters

### DISCUSSION THREAD

**Discussion by ErwinJapie** [fastapi/fastapi#6363] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class QueryArgs(BaseModel):
    x: str
    y: int

    class Config:
        extra = Extra.allow


app = FastAPI()

@app.get("/example")
async def example(query_args: QueryArgs = Depends()):
    return [ query_args.x, query_args.y, query_args.z ]
```

### Description

* Open the browser and call /example?x=test&y=3&z=extra_param
* Then expect response [ test, 3, extra_param ]

Currently this always returns a 422 status code no matter which parameters are supplied.

### Wanted Solution

Extra parameters are already possible for request bodies, it would be nice a feature to also support extra parameters for query params.

### Wanted Code

```python
class QueryArgs(BaseModel):
    x: str
    y: int

    class Config:
        extra = Extra.allow


app = FastAPI()

@app.get("/example")
async def example(query_args: QueryArgs = Depends()):
    return [ query_args.x, query_args.y, query_args.z ]
```

### Operating System

macOS

### FastAPI Version

0.68.2

### Python Version

3.9

**Comment by ddanier** [comment, marked as answer]
Having extra on the request body is - in my opinion - completely different from what is done when parsing the query parameters.

Think of the following code (base on your code):
```python
@app.get("/example")
async def example(query_args: QueryArgs = Depends(), some_other_arg: str):
    return [ query_args.x, query_args.y, query_args.z ]
```

Should `some_other_arg` now also be part of the `QueryArgs` model? If we find a solution for that: What happens if you use multiple models for query parameters that all allow extra fields - where should those go (first/last/all of them)? Please also keep in mind that other dependencies may also need to consume query parameters.

To be clear: I don't expect this to work in a sane way. If you want to implement this for your particular use case I would recommend using the `request` object directly. You could for example do something like:  
`query_args = QueryArgs(**request.query_params)` (untested - but you can definitely get this to work)  
(see https://fastapi.tiangolo.com/advanced/using-request-directly/ and https://www.starlette.io/requests/#query-parameters)

**Comment by ErwinJapie** [comment]
@ddanier thanks for the quick response.

I see what you're saying and that it is definitely not as straightforward as the extra parameter parsing for request bodies.

I will look into a solution using the request directly as you described.

**Comment by tiangolo** [comment]
Thanks for the help here @ddanier ! 👏  🙇 

Thanks for reporting back and closing the issue @ErwinJapie  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
